pkg/stateintospec/stateintospecdefaulter.go

```diff
@@ -55,6 +55,12 @@ func NewStateIntoSpecDefaulter(client client.Client) k8s.Defaulter {
 }
 
 func (v *StateIntoSpecDefaulter) ApplyDefaults(ctx context.Context, reconcilerType k8s.ReconcilerType, resource client.Object) (changed bool, err error) {
+	// Don't write the annotation for direct controllers
+	switch reconcilerType {
+	case k8s.ReconcilerTypeDirect:
+		return false, nil
```

Comment:
This seems correct but uninformative. Would be nice if when we ignored SIS:Merge that we logged that we were ignoring it.